Code/Source/Frame/ROS2FrameComponent.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "Frame/ROS2FrameComponent.h"
+#include "Utilities/ROS2Names.h"
+#include "ROS2/ROS2Bus.h"
+#include <AzCore/Component/Entity.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/EditContextConstants.inl>
+#include <AzCore/Serialization/SerializeContext.h>
+
+namespace ROS2
+{
+    void ROS2FrameComponent::Activate()
+    {
+        if (m_publishTransform)
+        {
+            m_ros2Transform = AZStd::make_unique<ROS2Transform>(GetParentFrameID(), GetFrameID(), IsDynamic());
+            if (IsDynamic())
+            {
+                AZ::TickBus::Handler::BusConnect();
+            }
+            else
+            {
+                m_ros2Transform->Publish(GetFrameTransform());
+            }
+        }
+        m_namespaceConfiguration.PopulateNamespace(IsTopLevel(), GetEntity()->GetName());
+    }
+
+    void ROS2FrameComponent::Deactivate()
+    {
+        if (m_publishTransform)
+        {
+            if (IsDynamic())
+            {
+                AZ::TickBus::Handler::BusDisconnect();
+            }
+            m_ros2Transform.reset();
+        }
+    }
+
+    void ROS2FrameComponent::OnTick([[maybe_unused]] float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint time)
+    {
+        m_ros2Transform->Publish(GetFrameTransform());
+    }
+
+    AZStd::string ROS2FrameComponent::GetGlobalFrameName()
+    {
+        // TODO - parametrize this, sometimes it is "world" and sometimes "map".
+        return "world";
+    }
```

Comment:
If you update the return value to be a `const char*`, it would provide a bit more flexibility for the caller.

That would allow the callers to store the value in a `std::string` if they chose.